apps/designer/app/canvas/features/text-editor/text-editor.stories.tsx

```diff
@@ -5,61 +5,86 @@ import { action } from "@storybook/addon-actions";
 import { Box } from "@webstudio-is/design-system";
 import { useSubscribe, publish } from "~/shared/pubsub";
 import { createInstance } from "~/shared/tree-utils";
+import type { TextToolbarState } from "~/designer/shared/nano-states";
 import { TextEditor } from "./text-editor";
 
 export default {
   component: TextEditor,
   title: "Text Editor 2",
 } as ComponentMeta<typeof TextEditor>;
 
+type Format =
+  | "bold"
+  | "italic"
+  | "superscript"
+  | "subscript"
+  | "link"
+  | "span"
+  | "clear";
+
 export const Basic: ComponentStory<typeof TextEditor> = ({ onChange }) => {
-  const [isEnabled, setIsEnabled] = useState(false);
-  const [isBold, setIsBold] = useState(false);
-  const [isItalic, setIsItalic] = useState(false);
-  const [isLink, setIsLink] = useState(false);
-  const [isSpan, setIsSpan] = useState(false);
+  const [state, setState] = useState<null | TextToolbarState>(null);
```

Comment:
could we use undefined as a default? Would it be bad for your semantics? Seems like the logic wouldn't care 🙏🙏🙏

Reply:
15 more characters 😢

Reply:
gzip ftw :)))

Reply:
My hands hurt when I write "undefined", ah damn ✋

Reply:
copilot )
or 
omit the value )))